csharp/route_guide/README.md

```diff
@@ -0,0 +1,392 @@
+            int lon = feature.Location.Longitude;
+            if (lon >= left && lon <= right && lat >= bottom && lat <= top)
+            {
+                await responseStream.WriteAsync(feature);
+            }
+        }
+    }
+```
+
+As you can see, here the request object is a `Rectangle` in which our client wants to find `Feature`s, but instead of returning a simple response we need to write responses to an asynchronous stream `IServerStreamWriter` using async method `WriteAsync`.
+
+Similarly, the client-side streaming method `RecordRoute` uses an [IAsyncEnumerator](https://github.com/Reactive-Extensions/Rx.NET/blob/master/Ix.NET/Source/System.Interactive.Async/IAsyncEnumerator.cs), to read the stream of requests using async method `MoveNext` and property `Current`.
+
+```csharp
+    public async Task<RouteSummary> RecordRoute(Grpc.Core.ServerCallContext context,
+	    Grpc.Core.IAsyncStreamReader<Point> requestStream)
+    {
+        int pointCount = 0;
+        int featureCount = 0;
+        int distance = 0;
+        Point previous = null;
+        var stopwatch = new Stopwatch();
+        stopwatch.Start();
+
+        while (await requestStream.MoveNext())
+        {
+            var point = requestStream.Current;
+            pointCount++;
+            if (RouteGuideUtil.Exists(CheckFeature(point)))
+            {
+                featureCount++;
+            }
+            if (previous != null)
+            {
+                distance += (int) CalcDistance(previous, point);
+            }
+            previous = point;
+        }
+
+        stopwatch.Stop();
+        return RouteSummary.CreateBuilder().SetPointCount(pointCount)
+            .SetFeatureCount(featureCount).SetDistance(distance)
+            .SetElapsedTime((int) (stopwatch.ElapsedMilliseconds / 1000)).Build();
+    }
+```
+Finally, let's look at our bidirectional streaming RPC `RouteChat`.
+
+```csharp
+    public async Task RouteChat(Grpc.Core.ServerCallContext context,
+	    Grpc.Core.IAsyncStreamReader<RouteNote> requestStream, Grpc.Core.IServerStreamWriter<RouteNote> responseStream)
+    {
+        while (await requestStream.MoveNext())
+        {
+            var note = requestStream.Current;
+            List<RouteNote> notes = GetOrCreateNotes(note.Location);
+
+			List<RouteNote> prevNotes;
+            lock (notes)
+            {
+                prevNotes = new List<RouteNote>(notes);
+            }
+
+            foreach (var prevNote in prevNotes)
+            {
+                await responseStream.WriteAsync(prevNote);
+            }                
+                
+            lock (notes)
+            {
+                notes.Add(note);
+            }
+        }
+    }
+```
+
+Here the method receives both `requestStream` and `responseStream` as an argument.  Reading the requests is done in a same way as in the `RecordRoute` example.  Writing the responses is done the same way as in the `ListFeatures` example.
```

Comment:
Reading the requests is done the same way as in the client-side streaming method `RecordRoute`. Writing the responses is done the same way as in the server-side streaming method `ListFeatures`..